### PR TITLE
Feat/trivia batch api

### PR DESCRIPTION
**Pull request by danieljacquin**
**Endpoint**
- Add POST /trivia/batch accepting up to 50 filters
- Each filter combines category and difficulty to fetch a random question
- Implements partial failure — failed filters return an error while successful ones still return results
- Results preserve the same order as the input filters

**Tests**
- Add service tests for all succeed and preserved order
- Add handler tests for happy path, empty filters, exceeds limit, and usage count header

**Docs**
- Add endpoint documentation
- update enspoints_count from 1 to 2

## Summary by CodeRabbit

* **New Features**
  * Added a batch trivia endpoint to fetch multiple questions with distinct filters in a single request; preserves per-filter category/difficulty and returns either a question or an error for each entry.
  * Supports up to 50 filters per batch.

* **Tests**
  * Added unit and HTTP tests covering success, ordering, validation errors, and batch size limits.

* **Documentation**
  * API docs updated with endpoint details and multi-language examples.

[Review Change Stack](https://app.coderabbit.ai/change-stack/bobadilla-tech/requiems-api/pull/665?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: apps/api/services/entertainment/trivia/service.go [hunk 1]
```diff
@@ -14,6 +14,16 @@ type Question struct {
...
 
+// BatchResponse represents the result of a single trivia query
+// within a batch request, pairing the original filter (category
+// and difficulty) with either a question or an error.
+type BatchResponse struct {
+	Category   string   `json:"category"`
+	Difficulty string   `json:"difficulty"`
+	Error      string   `json:"error,omitempty"`
+	Data       Question `json:"data"`
+}
```
[the remark this answers is not in the document]
Comment (danieljacquin) [reply apps/api/services/entertainment/trivia/service.go:25]: Response contract type is no longer located in type.go, it now lives in service.go
